### PR TITLE
feat: auto pick article on post card share new squad

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- auto pick an article when the share button on the post card is clicked then selected "Post to your Squad".

Note: In the preview below, I manually made the menu item available even when the user already has Squads.

https://user-images.githubusercontent.com/13744167/213085847-51570497-006b-460d-bf4d-a7b80f545fba.mov

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-922 #done

### CODE REVIEW

File: packages/shared/src/components/modals/NewSquadModal.tsx [hunk 1]
```diff
@@ -44,29 +47,32 @@ function NewSquadModal({
...
-    {
-      key: ModalState.WriteComment,
-    },
-    {
-      key: ModalState.Ready,
-      title: <Modal.Header.Title>{ModalState.Ready}</Modal.Header.Title>,
-    },
-  ];
+  const modalSteps = useMemo(() => {
+    const steps: ModalStep[] = [
```
Comment (el) [inline packages/shared/src/components/modals/NewSquadModal.tsx:51]: Here is a cleaner way to achieve this

```
  const modalSteps: ModalStep[] = [
    {
      key: ModalState.Details,
      title: (
        <>
          {onPreviousState && (
            <Modal.Header.StepsButton onClick={onPreviousState} />
          )}
          <Modal.Header.Subtitle>{ModalState.Details}</Modal.Header.Subtitle>
        </>
      ),
    },
    !post ? {
      key: ModalState.SelectArticle,
    } : undefined,
    {
      key: ModalState.WriteComment,
    },
    {
      key: ModalState.Ready,
      title: <Modal.Header.Title>{ModalState.Ready}</Modal.Header.Title>,
    },
  ].filter(step => step);
```

**Review by el — CHANGES_REQUESTED** [review]
Just a minor request, other than that, lgtm ❤️

**Review by rebelchris — APPROVED** [review]
Besides @el comment nothing blocking from my side.

File: packages/shared/src/components/modals/NewSquadModal.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/NewSquadModal.tsx:51]: Nice! Love this refactor! Let me update it quickly!

**Pull request merged**